### ISSUE TITLE
Various issues

### ISSUE THREAD

**Issue by JPfeP** [robertesler/cordova-pd#3]
Hi, 

Thanks for the new version, I have a few usage questions/issues under Android:

1) I can edit the cordova.pd file and hear the change but when I try to invoke abstractions contained in other Pd files located  in the same "res/raw" dir, it seems there are not found by the Pd thread. Yet, if I verify in the APK archive, the Pd files are well copied and present. I tried to put the abstractions in various dirs and made some attempts with the "declare -path" block but without success, I am just clueless right now where to put them and how to use a proper relative path. 

2) I tried to play a sample file in the cordova.pd patch and naively put the wav sound in the same dir in "res/raw" and it doesn't work for various reasons (at first the file named 'CH.wav' was not allowed due to the capital letters, then once renamed, was again not accepted with an error message like "Stream output broken" (or so, from memory, sorry). Directories are not allowed as well here, seemingly due to their nature. Tried to put the sample waw file elsewhere, but without success, similarly to the first issue. Didn't try absolute path, however.    

3) With the last version of the plugin I have a little problem, Cordova doesn't fire the "Device ready" event anymore, an debugging with Chrome shows this:
" Uncaught module exec not found" (in red) 
" Ionic Native: deviceready did not fire within 5000ms. This can happen when plugins are in an inconsistent state. Try removing plugins from plugins/ and reinstalling them."
Consequently the splash-screen stays forever if you have set it to hide on that event. It was working OK with the previous version of your plugin and removing the plugin "cures" the issue. 

4) BTW, I still have the 404 error message but it is in fact harmless.

**Comment by robertesler** [comment]
Do you have your project online somewhere?  Like GitHub or similar?  I could look at it and see what’s happening.   None this sounds like it is specific to the plugin but rather some issue with the libpd dependency. But I can’t really tell without seeing your project’s code. 
Thanks.

**Comment by JPfeP** [comment]
Issue #3 might be related to a change you made in cordova-pd.js recently :

L12:  +var exec = require('exec');

It's a an external npm module, right ? This may be : https://www.npmjs.com/package/exec

The page says it's a deprecated module, should I install it nonetheless ? May be you should warn the users in the documentation if it is a requirement.

**Comment by robertesler** [comment]
Ok.  I don't have time today to fix any of these issues.  Try commenting out that line of code and see if fixes any of your issue.  Cordova plugins can break things so I can't track that issue specifically.  As for the Pd abstractions that looks like a libpd issue and I don't know the solution right now.  I will make a ticket on their github site.   
The 404 issue, do you still have the old "PdPlugin" line in your config.xml?  If so remove it and make sure something like:
`<plugin name="org.urbanstew.cordova.pd" spec="0.0.2-dev" />`
exists.
Also, send me your project code, that will help me know the precise issues you are dealing with and to assess whether they are my problem or someone else's.  
I'll close this issue if any changes are made.
THank you.

**Comment by JPfeP** [comment]
Thanks for your answer,

I can remove the old PdPlugin line but each time I build the project it reappears. And automatically is added as well this one: 
`<plugin name="org.urbanstew.cordova.pd" spec="git+https://github.com/robertesler/cordova-pd.git" />`

Anyway, not a real problem. Purely cosmetic... But if you cannot reproduce it, may be there are some old files kept on my project dir despite the upgrade. I will start a fresh new test project to get sure. Perhaps a problem with my current version of Cordova as well (or another component) ? I will try to see if there are some updates. 

I can indeed provide you this test project once it's done. 

BTW, I tried in the meanwhile to install the exec npm module but it doesn't solve the related issue (Uncaught module exec not found), which puzzle me a little bit : it's yet installed in my nodes_modules directory now.

The other solution, commenting the "require('exec')" line, as you suggested, has however solved the problem, the splash-screen hides as before. 

Edit: No improvement on issue #1

**Comment by JPfeP** [comment]
Just a quick word to say that issue #4 is solved here, there was a deprecated reference to "PdPlugin" in the "package.json" file of my project. I removed it manually and now it's fine.

**Comment by JPfeP** [comment]
Thanks for the update, it saves my day ! ;)

All issues are solved now, I am very happy to hear my patches on my smartphone. That's so cool. 

I was trying to fix the abstractions problem by myself using PdBase.addToSearchPath() but I realize now that probably there are some security restrictions in the context of an activity and that was not successful.
Using a zip file seems to lift automatically theses restrictions, it's fine for now and allow me to progress on the whole app. Will see later if a user needs to access external storage.

I had to root my phone to investigate theses issues, so it was the occasion to learn something.

BTW, on my local version of the package.json file from your node module, I have added this line:
` "main": "www/cordova-pd.js",`

This allows to import more elegantly the module like this:
`import Pd from 'org.urbanstew.cordova.pd';`

instead of using (<any>window).plugins.pd which doesn't seems to work anymore on top of that in the current version of Ionic.
